src/Conversion/ONNXToMhlo/NN/Conv.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===----------- Normalization.cpp - Lowering Normalization Ops -----------===//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers ONNX Conv Operators to Mhlo dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToMhlo/ONNXToMhloCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+#include "src/Support/TypeUtilities.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+struct ONNXConvOpLoweringToMhlo : public ConversionPattern {
+  ONNXConvOpLoweringToMhlo(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXConvOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+
+    ONNXConvOpAdaptor operandAdaptor(operands);
+    ONNXConvOp convOp = llvm::dyn_cast<ONNXConvOp>(op);
+    Location loc = op->getLoc();
+
+    ONNXConvOpShapeHelper shapeHelper(&convOp);
```

Comment:
I remember that when I tried to use shapehelper for sliceOp, fGetDenseVal failed since we only pass the operator pointer to the shapeHelper, but I couldn't remember clearly what the issue is.